core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -2292,7 +2292,27 @@ private void parseSelectClause(GenericLexer lexer, QueryModel model, SqlParserCa
                 QueryColumn col;
                 final int colPosition = lexer.lastTokenPosition();
 
-                if (tok != null && isOverKeyword(tok)) {
+                // support [first_value | last_value] (expr) [ignore nulls] over(...) window function
```

Comment:
rather than rewriting ast we can save "ignore nulls" requirement into the window column. "ignore nulls" is not a valid syntax without `over` anyway.


Once in the window column, "ignore nulls" should be made available to the function factory and the factory (which is mostly similar between not_null and regular) will decide which funciton to generate based on window context

Reply:
I think we should not expose `_not_null` to user, e.g. even if you use `EXPLAIN` the functions should serialize as:

```
first_value(x) ignore nulls over(...)
```

rather than

```
first_non_null_value(x) over(...)
```